Rectangle.js

```diff
@@ -0,0 +1,157 @@
+class Rectangle {
+
+    constructor(x, y, width = 150, height = 80, color = "#800080") {
+        this.x = x;
+        this.y = y;
+        this.width = width;
+        this.height = height;
+        this.color = color;
+        this.collided = false;
+        this.startCoordinates = {x: x, y: y};
+        this.stickySide = null;
+        this.stickedTo = null;
+    }
+
+    findDistance(first, second, firstCenter, secondCenter, deviation, greater) {
+        let distance = (first / 2 + second / 2 + deviation);
+        let currentDistance = Math.abs(firstCenter - secondCenter);
+        return greater ? currentDistance > distance : currentDistance < distance;
+    }
+
+    checkCollision(first, second) {
+        let firstCenterX = first.x + first.width / 2;
+        let firstCenterY = first.y + first.height / 2;
+        let secondCenterX = second.x + second.width / 2;
+        let secondCenterY = second.y + second.height / 2;
+        let collideX = this.findDistance(first.width, second.width, firstCenterX, secondCenterX, 0, false);
+        let collideY = this.findDistance(first.height, second.height, firstCenterY, secondCenterY, 0, false);
+        return collideX && collideY;
+    }
+
+    checkSticky(first, second) {
+        let firstCenterX = first.x + first.width / 2;
+        let firstCenterY = first.y + first.height / 2;
+        let secondCenterX = second.x + second.width / 2;
+        let secondCenterY = second.y + second.height / 2;
+        let stickyX = this.findDistance(first.width, second.width, firstCenterX, secondCenterX, N, false);
+        let stickyY = this.findDistance(first.height, second.height, firstCenterY, secondCenterY, N, false);
+        let collideX = this.findDistance(first.width, second.width, firstCenterX, secondCenterX, 0, true);
```

Comment:
Add an early return after stickyY? 
```
if (!stickyX || !stickyY) {
    return null;
}
```
It will eliminate unnecessary calculations, and will also reduce the length of conditions below by removing stickyX and stickyY checks